modules/bibauthorid/lib/bibauthorid_webinterface.py

```diff
@@ -3724,7 +3726,7 @@ def __call__(self, req, form):
             try:
                 pid = int(self.path)
             except ValueError:
-                redirect_to_url(req, "%s/author/search?q=%s" % (CFG_BASE_URL, self.path))
+                redirect_to_url(req, "%s/author/search?q=%s" % (CFG_BASE_URL, urllib.quote(self.path)))
```

Comment:
I have double checked this one and `self.path` is populated internally by BibAuthorID and is always indeed a string or `None` but in this latter case it is captured earlier in the function.